kuma/users/tests/test_views.py

```diff
@@ -388,10 +390,119 @@ def test_post_submits_no_revisions_to_akismet_when_wrong_revisions_in_request(se
+        eq_(new_document.current_revision.id, latest_revision.id)
+        # Admin is the creator of this current revision
+        eq_(new_document.current_revision.creator, self.admin)
+
+    def test_post_one_reverts_one_does_not_revert(self):
+        """POSTing to ban_user_and_cleanup url with revisions to 2 documents."""
+        # Document A will have latest revision by the admin, but older spam revisions
+        # Document B will have latest revision by spammer
+        # Document A should not revert (although there are older spam revisions)
+        # Document B will revert
+        new_document_a = create_document(save=True)
+        new_document_b = create_document(save=True)
+        self.create_revisions(
+            num=1, document=new_document_a, creator=self.admin)
+        self.create_revisions(
+            num=1, document=new_document_b, creator=self.admin)
+        spam_revisions_a = self.create_revisions(
+            num=3,
+            document=new_document_a,
+            creator=self.testuser)
+        safe_revision_a = self.create_revisions(
+            num=1,
+            document=new_document_a,
+            creator=self.admin)
+        spam_revisions_b = self.create_revisions(
+            num=3,
+            document=new_document_b,
+            creator=self.testuser)
+
+        # Pass in all spam revisions:
+        # A revisions will not be reverted
+        # B revisions will be reverted
+        data = {'revision-id': [rev.id for rev in spam_revisions_a + spam_revisions_b]}
+
+        self.client.login(username='admin', password='testpass')
+        resp = self.client.post(self.ban_testuser_url, data=data)
+        eq_(200, resp.status_code)
+
+        # Document A: No changes should have been made
+        new_document_a = Document.objects.filter(id=new_document_a.id).first()
+        eq_(new_document_a.current_revision.id, safe_revision_a[0].id)
+        revisions_a = Revision.objects.filter(document=new_document_a)
+        eq_(revisions_a.count(), 5)  # Total of 5 revisions, no new revisions were made
+
+        # Document B: Make sure that the current revision is not the spam revision
+        new_document_b = Document.objects.filter(id=new_document_b.id).first()
+        for revision in spam_revisions_b:
+            ok_(revision.id != new_document_b.current_revision.id)
+        # The most recent Revision for this document
+        # should be the document's current revision
+        latest_revision_b = Revision.objects.filter(
+            document=new_document_b).order_by('-id').first()
+        eq_(new_document_b.current_revision.id, latest_revision_b.id)
+        # Admin is the creator of this current revision
+        eq_(new_document_b.current_revision.creator, self.admin)
+        revisions_b = Revision.objects.filter(document=new_document_b)
+        # 7 revisions on B = 1 initial + 3 spam revisions + 3 new revert revisions
+        eq_(revisions_b.count(), 7)
```

Comment:
Please add a check that ``new_document_b.current_revision.content`` is the same as the initial revision's content, to verify the revert was correct.